apps/core/app/[locale]/not-found.tsx

```diff
@@ -30,23 +37,29 @@ export default async function NotFound() {
       />
       <main className="mx-auto mb-10 max-w-[835px] space-y-8 px-6 sm:px-10 lg:px-0">
         <Message className="flex-col gap-8 px-0 py-16">
-          <h2 className="text-4xl font-black lg:text-5xl">We couldn't find that page!</h2>
-          <p className="text-lg">
-            It looks like the page you requested has moved or no longer exists.
-          </p>
+          <h2 className="text-4xl font-black lg:text-5xl">{t('heading')}</h2>
+          <p className="text-lg">{t('message')}</p>
         </Message>
-        <SearchForm />
+        <NextIntlClientProvider locale={locale} messages={pick(dictionary, 'NotFound')}>
+          <SearchForm />
+        </NextIntlClientProvider>
         <section>
-          <h3 className="mb-8 text-3xl font-black lg:text-4xl">Featured products</h3>
+          <h3 className="mb-8 text-3xl font-black lg:text-4xl">{t('featuredProducts')}</h3>
           <div className="grid grid-cols-2 gap-x-8 gap-y-8 md:grid-cols-4">
             {featuredProducts.map((product) => (
-              <ProductCard
+              <NextIntlClientProvider
                 key={product.entityId}
-                product={product}
-                showCart={false}
-                showCompare={false}
-                showReviews={false}
-              />
+                locale={locale}
+                messages={pick(dictionary, 'Product')}
+              >
+                <ProductCard
+                  key={product.entityId}
```

Comment:
I don't think we need a key here anymore if inside Provider.

Reply:
Yeah, i've missed it here.